pkg/tkctl/cmd/version/version.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019. PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package version
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/pingcap/tidb-operator/pkg/label"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/config"
+	"github.com/pingcap/tidb-operator/version"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/apis/core"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+const (
+	versionExample = `
+		# Print the cli version and tidb operator version
+		tkctl version
+`
+)
+
+type VersionOptions struct {
+	io.Writer
+}
+
+func NewCmdVersion(tkcContext *config.TkcContext, out io.Writer) *cobra.Command {
+	options := &VersionOptions{
+		Writer: out,
+	}
+	cmd := &cobra.Command{
+		Use:     "version",
+		Short:   "Print the client & server version",
+		Example: versionExample,
+		Run: func(_ *cobra.Command, _ []string) {
+			cmdutil.CheckErr(options.runVersion(tkcContext))
+		},
+	}
+
+	return cmd
+}
+
+func (o *VersionOptions) runVersion(tkcContext *config.TkcContext) error {
+	restConfig, err := tkcContext.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	kubeCli, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	clientVersion := version.Get()
+
+	controllers, err := kubeCli.AppsV1().
+		Deployments(core.NamespaceAll).
+		List(v1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s,%s=%s", label.ComponentLabelKey, "controller-manager", label.InstanceLabelKey, "tidb-operator"),
+		})
+	if err != nil {
+		return err
+	}
+	schedulers, err := kubeCli.AppsV1().
+		Deployments(core.NamespaceAll).
+		List(v1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s,%s=%s", label.ComponentLabelKey, "scheduler", label.InstanceLabelKey, "tidb-operator"),
+		})
+	if err != nil {
+		return nil
+	}
+
+	fmt.Fprintf(o, "Client Version: %s\n", clientVersion.GitVersion)
```

Comment:
Use `clientVersion` is enough

Reply:
Brilliant, you have all the codes in your brain I suppose...